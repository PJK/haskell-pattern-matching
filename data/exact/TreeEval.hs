module TreeEval where

data Tree
  = Leaf
  | Node Tree Tree


func' :: Tree -> Tree -> Int
func' _             Leaf = 1
func' (Node Leaf _) Leaf = 2
func' _             _    = 3
