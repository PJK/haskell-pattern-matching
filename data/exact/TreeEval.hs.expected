{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "func'",
            {
                "tag": "InaccessibleRhs",
                "contents": [
                    {
                        "tag": "ConstructorPattern",
                        "contents": [
                            "Node",
                            [
                                {
                                    "tag": "ConstructorPattern",
                                    "contents": [
                                        "Leaf",
                                        []
                                    ]
                                },
                                {
                                    "tag": "WildcardPattern",
                                    "contents": []
                                }
                            ]
                        ]
                    },
                    {
                        "tag": "ConstructorPattern",
                        "contents": [
                            "Leaf",
                            []
                        ]
                    }
                ]
            }
        ]
    ]
}
